docs/5-design-patterns/homework/package.json

```diff
@@ -0,0 +1,19 @@
+{
+  "name": "homework",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "ts-node homework.ts",
+    "build": "tsc",
+    "test": "jest",
+    "test:fp": "jest test/fp"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "ts-node": "^10.9.1",
+    "typescript": "^4.8.4"
+  }
+}
```

Comment:
I would recommend you a very good resource for design patterns https://refactoring.guru/ru/design-patterns/decorator/typescript/example

MR is very good, I've checked your realisation of patterns and it is in right way.